### PR TITLE
Add a section on ternary operators

### PR DESCRIPTION
**Pull request by fchorney**
Does this seem sane?

Do we need more sections on other boolean expressions of any sort?

### CODE REVIEW

**Comment by nickrobinson251** [comment]
Perhaps make explicit never nest them i.e. `x == 1 ? a : x == 2 ? b : c`

**Comment by nickrobinson251** [comment]
This is already our style; not sure why we didn't write it down sooner. Thanks for adding!

File: README.md [hunk 1]
```diff
@@ -622,6 +623,30 @@ Floating-point numbers should always include a leading and/or trailing zero:
 3.f0
 ```
 
+### Ternary Operator
+
+Ternary operators should generally only consume a single line.
```
Comment (nickrobinson251) [inline README.md]:
```suggestion
Ternary operators should generally only consume a single line.
Do not chain multiple ternary operators.
```

File: README.md [hunk 2]
```diff
@@ -622,6 +623,30 @@ Floating-point numbers should always include a leading and/or trailing zero:
...
+
+```julia
+# Yes:
+foobar = foo == 2 ? bar : baz
+
+# No:
+foobar = foo == 2 ?
+    bar :
+    baz
```
Comment (nickrobinson251) [inline README.md]:
```suggestion
    baz
foobar = foo == 2 ? bar : foo == 3 ? qux : baz
```

File: README.md (hunk 1 above)
Comment (oxinabox) [reply README.md]: I am dubious sometimes it is the best way.
Maybe: Think twice before chaining ternery operators,
and consider if using dispatch, or a dictionary might be better,

Comment (nickrobinson251) [reply README.md]: well, those seem like performance questions.

style-wise (i.e. when what you want is 1 `if ... elseif ... elseif ...`) we do in fact always say 
```
# Yes:
foobar = if foo == 2
    bar
elseif foo == 3
    qux
else
    baz
end

# No:
foobar = foo == 2 ? bar : foo == 3 ? qux : baz
```

Comment (nickrobinson251) [reply README.md]: Is there a performance difference between the two examples I give?

Comment (nickrobinson251) [reply README.md]: Or is your point about clarity when there are many conditions?

We could add: "If chaining many conditions, consider if using dispatch or a dictionary would be clearer."

Comment (nickrobinson251) [reply README.md]: Happy to discuss wording/examples here, but if we don't think this is already a settled matter then let's leave it to another PR, and merge this one as-is. :)

Comment (oxinabox) [reply README.md]: You are correct.

Comment (nickrobinson251) [reply README.md]: How's this?

```suggestion
Ternary operators should generally only consume a single line.
Do not chain multiple ternary operators.
If chaining many conditions, consider using dispatch or a dictionary.
```

Comment (oxinabox) [reply README.md]: fine

Comment (oxinabox) [reply README.md]: Or even leave it with your original suggestion

Comment (nickrobinson251) [reply README.md]: 👍

**Comment by fchorney** [comment]
This good to merge now? Any other thoughts?

File: README.md [hunk 3]
```diff
@@ -622,6 +623,33 @@ Floating-point numbers should always include a leading and/or trailing zero:
 3.f0
 ```
 
+### Ternary Operator
+
+Ternary operators should generally only consume a single line.
+Do not chain multiple ternary operators.
+If chaining many conditions, consider using dispatch or a dictionary.
```
Comment (omus) [inline README.md]:
```suggestion
If chaining many conditions, consider using an `if`-`elseif`-`else` conditional, dispatch, or a dictionary.
```

File: README.md (hunk 1 above)
Comment (omus) [reply README.md]: Keeps this inline with how the official docs talk about this:  https://docs.julialang.org/en/v1/manual/control-flow/#Control-Flow-1
```suggestion
Ternary operators (`?:`) should generally only consume a single line.
```

File: README.md [hunk 4]
```diff
@@ -622,6 +623,33 @@ Floating-point numbers should always include a leading and/or trailing zero:
...
+
+As an alternative, you can use a compound boolean expression:
+
+```julia
+foobar = if foo == 2
+    bar
+else
+    baz
+end
+```
```
Comment (omus) [inline README.md]: Maybe add the equivalent to the chained ternary here?

**Pull request merged**